OTP-P-Sample/OTPServer.p

```diff
@@ -0,0 +1,100 @@
+event Success;
+event Failure;
+
+event OTPSecretMsg: (machine, int);
+event OTPSecretReceived;
+event OTPCodeMsg: int;
+event OTPCodeValidated;
+event OTPCodeFailed;
+
+machine BANK_SERVER 
+{
+    var clientOtpGenerator: machine;
+	var secret: int;
+
+    // This is the entry point.
+    start state Init {
+        entry (payload:any) {
+          clientOtpGenerator = payload as machine;
+          raise (Success);   	   
+        }
+        on Success goto GenerateOTPSecret;
+    }
+
+    state GenerateOTPSecret {
+        entry {
+			// generate OTP secret 
+			secret = 123456788;
+			send clientOtpGenerator, OTPSecretMsg, (this, secret);
+	    }
+        on OTPSecretReceived goto WaitOTPCode;
+     }
+
+     state WaitOTPCode {
+        on OTPCodeMsg goto ValidateOTPCode;
+     }
+
+	 state ValidateOTPCode {
+        entry (payload: int) {
+		  // validate OTP code
+		  if (payload == 123456789) {
+			send clientOtpGenerator, OTPCodeValidated;
+          	raise (Success);  
```

Comment:
here also you can delete the success and done events and replace raise by direct jumps to states using goto.